code/go/0chain.net/smartcontract/storagesc/challenge.go

```diff
@@ -1116,7 +1117,11 @@ func (sc *StorageSmartContract) addChallenge(alloc *StorageAllocation,
 	for _, id := range expiredIDs {
 		_, err := balances.DeleteTrieNode(storageChallengeKey(sc.ID, id))
 		if err != nil {
-			return common.NewErrorf("add_challenge", "could not delete challenge node: %v", err)
+			if err == util.ErrValueNotPresent {
+				lenExpiredIDs--
```

Comment:
We don't need to do this check, we should not have ErrValueNotPresent otherwise we need to return error instead of continue.

Reply:
When this happen it blocks challenge generation where expiredIDs contain some duplicates (still trying to figure out why this happens).